test/invariants/fuzz/BMath.t.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.25;
+
+import {EchidnaTest} from '../helpers/AdvancedTestsUtils.sol';
+import {BMath} from 'contracts/BMath.sol';
+
+contract FuzzBMath is EchidnaTest {
+  BMath bMath;
+
+  uint256 immutable MIN_WEIGHT;
+  uint256 immutable MAX_WEIGHT;
+  uint256 immutable MIN_FEE;
+  uint256 immutable MAX_FEE;
+
+  /**
+   * NOTE: These values were chosen to pass the fuzzing tests
+   * @dev Reducing BPOW_PRECISION may allow broader range of values increasing the gas cost
+   */
+  uint256 constant MAX_BALANCE = 1_000_000e18;
+  uint256 constant MIN_BALANCE = 1e18;
+  uint256 constant MAX_TOLERANCE = 1e12;
+
+  constructor() {
+    bMath = new BMath();
+
+    MIN_WEIGHT = bMath.MIN_WEIGHT();
+    MAX_WEIGHT = bMath.MAX_WEIGHT();
+    MIN_FEE = bMath.MIN_FEE();
+    MAX_FEE = bMath.MAX_FEE();
+  }
+
+  // calcOutGivenIn should be inverse of calcInGivenOut
+  function testCalcInGivenOut_InvCalcInGivenOut(
+    uint256 tokenBalanceIn,
+    uint256 tokenWeightIn,
+    uint256 tokenBalanceOut,
+    uint256 tokenWeightOut,
+    uint256 tokenAmountIn,
+    uint256 swapFee
+  ) public view {
+    tokenWeightIn = clamp(tokenWeightIn, MIN_WEIGHT, MAX_WEIGHT);
+    tokenWeightOut = clamp(tokenWeightOut, MIN_WEIGHT, MAX_WEIGHT);
+    tokenAmountIn = clamp(tokenAmountIn, MIN_BALANCE, MAX_BALANCE);
+    tokenBalanceOut = clamp(tokenBalanceOut, MIN_BALANCE, MAX_BALANCE);
+    tokenBalanceIn = clamp(tokenBalanceIn, MIN_BALANCE, MAX_BALANCE);
+    swapFee = clamp(swapFee, MIN_FEE, MAX_FEE);
+
+    uint256 calc_tokenAmountOut =
+      bMath.calcOutGivenIn(tokenBalanceIn, tokenWeightIn, tokenBalanceOut, tokenWeightOut, tokenAmountIn, swapFee);
+
+    uint256 calc_tokenAmountIn =
+      bMath.calcInGivenOut(tokenBalanceIn, tokenWeightIn, tokenBalanceOut, tokenWeightOut, calc_tokenAmountOut, swapFee);
+
+    assert(
```

Comment:
I'm not getting any coverage on this assert, all calls seem to revert on the previous line